CMakeLists.txt

```diff
@@ -12,6 +12,10 @@ ament_auto_find_build_dependencies()
 # Build
 ##############################################################################
 
+if($ENV{ROS_DISTRO} STREQUAL "humble")
+    add_compile_definitions(IS_HUMBLE)
+endif()
```

Comment:
This works, but I think what we actually want is to check the version of the rclcpp package. Cmake sets version variables after they are find_package'd; how about checking that '${rclcpp_VERSION_MAJOR}' is greater than or equal to 20? (it looks like matched callbacks were added in version 20)

Reply:
Also it would be clearer to call the compile define RCLCPP_SUPPORTS_MATCHED_CALLBACKS

Reply:
I was looking for a version macro in rclcpp package and couldn't find one. I was not aware of version variables set by cmake. I applied your suggestions.
btw. a lot of ROS packages don't use a consistent versioning scheme but rclcpp is a quality level 1 package so I have more hope.

Reply:
True that there's no strict versioning scheme across ROS, but I think we can at least expect the major version to to _decrease_ :-)